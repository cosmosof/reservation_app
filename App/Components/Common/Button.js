// @flow

import React from 'react';
import {TouchableHighlight, Text} from 'react-native';
import styles from './Styles/ButtonStyles';

type Props = {
  onPress: Function,
  children?: string,
  disabled?: boolean,
  styles?: Object,
  textStyle?: Object,
  buttonStyle?: Object,
  title?: string,
};

export default function Button(props: Props) {
  const {container, disabledStyle, text} = styles;
  const {buttonStyle, children, onPress, textStyle, title, disabled} = props;

  return (
    <TouchableHighlight
      disabled={disabled}
      onPress={onPress}
      style={[container, buttonStyle, disabled && disabledStyle]}
      underlayColor="transparent"
    >
      <Text style={[text, textStyle]}>{children || title || ''}</Text>
    </TouchableHighlight>
  );
}

Button.defaultProps = {
  buttonStyle: undefined,
  children: undefined,
  disabled: false,
  styles: undefined,
  textStyle: undefined,
  title: undefined,
};
