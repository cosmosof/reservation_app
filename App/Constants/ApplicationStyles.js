import {Platform} from 'react-native';
import Fonts from './Fonts';
import Colors from './Colors';

const ApplicationStyles = {
  navbarHeader: {
    color: Colors.darkMatBlue2,
    fontSize: 18,
    fontWeight: '600',
    fontFamily: Platform.OS == 'ios' ? Fonts.typeIOS.condensed : Fonts.typeAndroid.condensed,
  },
};

export default {
  ApplicationStyles,
};
