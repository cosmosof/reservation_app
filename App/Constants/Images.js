const images = {
  logo: require('../Images/Relate.png'),
};

export default images;
