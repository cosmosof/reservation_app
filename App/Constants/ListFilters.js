export default [
  {
    index: 0,
    name: 'guest name',
    value: 'name',
  },
  {
    index: 1,
    name: 'in guest names',
    value: 'name_contains',
  },
  {
    index: 3,
    name: 'name starts with',
    value: 'name_starts_with',
  },
  {
    index: 4,
    name: 'hotel name',
    value: 'hotelName',
  },
  {
    index: 5,
    name: 'in hotel names',
    value: 'hotelName_contains',
  },
  {
    index: 7,
    name: 'hotel names start with',
    value: 'hotelName_starts_with',
  },
  {
    index: 9,
    name: 'arrival date',
    value: 'arrivalDate',
  },
  {
    index: 10,
    name: 'departure date',
    value: 'departureDate',
  },
];
