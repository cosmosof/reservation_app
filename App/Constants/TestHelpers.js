const noop = () => {};
const sampleText = 'Sample text';
export default {
  noop,
  sampleText,
};
