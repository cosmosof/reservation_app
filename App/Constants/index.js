import Colors from './Colors';
import Fonts from './Fonts';
import Metrics from './Metrics';
import Images from './Images';
import ListFilters from './ListFilters';
import ApplicationStyles from './ApplicationStyles';
import TestHelpers from './TestHelpers';

export {Colors, Fonts, Images, Metrics, ListFilters, ApplicationStyles, TestHelpers};
