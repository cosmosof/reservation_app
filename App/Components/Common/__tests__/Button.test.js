import React from 'react';
import {shallow} from 'enzyme';
import Button from '../Button';
import {TestHelpers} from '../../../Constants';

const {noop, sampleText} = TestHelpers;

describe('<Button>', () => {
  const Component = <Button onPress={noop}>{sampleText}</Button>;
  describe('Structure', () => {
    it('renders correctly', () => {
      const wrapper = shallow(Component);
      expect(wrapper).toMatchSnapshot();
    });
    it('renders correctly if title prop is set', () => {
      const Clone = React.cloneElement(Component, {title: sampleText});
      const wrapper = shallow(Clone);
      expect(wrapper).toMatchSnapshot();
    });
    it('renders correctly if buttonStyle prop is set', () => {
      const Clone = React.cloneElement(Component, {buttonStyle: {backgroundColor: 'red'}});
      const wrapper = shallow(Clone);
      expect(wrapper).toMatchSnapshot();
    });
    it('renders correctly if styles prop is set', () => {
      const Clone = React.cloneElement(Component, {styles: {backgroundColor: 'blue'}});
      const wrapper = shallow(Clone);
      expect(wrapper).toMatchSnapshot();
    });
    it('renders correctly if textStyle prop is set', () => {
      const Clone = React.cloneElement(Component, {textStyle: {fontSize: 16}});
      const wrapper = shallow(Clone);
      expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly if disabled prop is set true', () => {
      const Clone = React.cloneElement(Component, {disabled: true});
      const wrapper = shallow(Clone);
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('Functionality', () => {
    it('handles onPress callback', () => {
      const onPressAgent = jest.fn();
      const Clone = React.cloneElement(Component, {onPress: onPressAgent});
      const wrapper = shallow(Clone);
      wrapper.simulate('press');
      expect(onPressAgent).toHaveBeenCalledTimes(1);
    });
  });
});
