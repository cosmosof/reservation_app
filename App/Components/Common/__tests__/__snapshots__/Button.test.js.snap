// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> Structure renders correctly 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  delayPressOut={100}
  disabled={false}
  onPress={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "gray",
        "maxHeight": 40,
        "padding": 8,
      },
      undefined,
      false,
    ]
  }
  underlayColor="transparent"
>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontFamily": "AvenirNextCondensed-DemiBold",
          "fontSize": 16,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    Sample text
  </Text>
</TouchableHighlight>
`;

exports[`<Button> Structure renders correctly if buttonStyle prop is set 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  delayPressOut={100}
  disabled={false}
  onPress={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "gray",
        "maxHeight": 40,
        "padding": 8,
      },
      Object {
        "backgroundColor": "red",
      },
      false,
    ]
  }
  underlayColor="transparent"
>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontFamily": "AvenirNextCondensed-DemiBold",
          "fontSize": 16,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    Sample text
  </Text>
</TouchableHighlight>
`;

exports[`<Button> Structure renders correctly if disabled prop is set true 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  delayPressOut={100}
  disabled={true}
  onPress={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "gray",
        "maxHeight": 40,
        "padding": 8,
      },
      undefined,
      Object {
        "backgroundColor": "#E2E6E8",
      },
    ]
  }
  underlayColor="transparent"
>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontFamily": "AvenirNextCondensed-DemiBold",
          "fontSize": 16,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    Sample text
  </Text>
</TouchableHighlight>
`;

exports[`<Button> Structure renders correctly if styles prop is set 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  delayPressOut={100}
  disabled={false}
  onPress={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "gray",
        "maxHeight": 40,
        "padding": 8,
      },
      undefined,
      false,
    ]
  }
  underlayColor="transparent"
>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontFamily": "AvenirNextCondensed-DemiBold",
          "fontSize": 16,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    Sample text
  </Text>
</TouchableHighlight>
`;

exports[`<Button> Structure renders correctly if textStyle prop is set 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  delayPressOut={100}
  disabled={false}
  onPress={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "gray",
        "maxHeight": 40,
        "padding": 8,
      },
      undefined,
      false,
    ]
  }
  underlayColor="transparent"
>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontFamily": "AvenirNextCondensed-DemiBold",
          "fontSize": 16,
          "textAlign": "center",
        },
        Object {
          "fontSize": 16,
        },
      ]
    }
  >
    Sample text
  </Text>
</TouchableHighlight>
`;

exports[`<Button> Structure renders correctly if title prop is set 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  delayPressOut={100}
  disabled={false}
  onPress={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "gray",
        "maxHeight": 40,
        "padding": 8,
      },
      undefined,
      false,
    ]
  }
  underlayColor="transparent"
>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontFamily": "AvenirNextCondensed-DemiBold",
          "fontSize": 16,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    Sample text
  </Text>
</TouchableHighlight>
`;
